lib/synapse_pay_rest/http_client.rb

```diff
@@ -87,13 +109,23 @@ def update_headers(oauth_key: nil, fingerprint: nil, client_id: nil,
     # 
     # @return [Hash] API response
     def post(path, payload, **options)
+      Rails.logger.info('-- Request: POST -----------') if @logging
       headers = get_headers
+      Rails.logger.info("Headers: #{headers}") if @logging
+
       if options[:idempotency_key]
         headers = headers.merge({'X-SP-IDEMPOTENCY-KEY' => options[:idempotency_key]})
       end
 
-      response = with_error_handling { RestClient::Request.execute(:method => :post, :url => full_url(path), :payload => payload.to_json, :headers => headers, :timeout => 300) }
-      p 'RESPONSE:', JSON.parse(response) if @logging
+      response = RestClient::Request.execute(method: :post,
```

Comment:
with_error_handling removed on purpose here?

Reply:
My bad. It was suppressing debugging info.